### PR TITLE
Distribution handshake cleanup / clarification

### PR DESCRIPTION
**Pull request by RaimoNiskanen**
This PR does a cleanup of the distribution handshake cookie handling code to better express the effective behaviour, adds test cases to consolidate that behaviour, and clarifies the documentation ~~(soon to be committed)~~.

It also adds useful functionality for handling differing cookies on different nodes:
* A command line argument `-setcookie Node Cookie`
* `erlang:get_cookie(Node)` ~~(soon to be committed)~~
* `erlang:set_cookie(Cookie)` ~~(soon to be committed)~~

### CODE REVIEW

**Comment by bjorng** [comment]
There is no extended error information for the new `set_cookie/1` and `get_cookie/1` BIFs.

Error information should be added to `lib/kernel/src/erl_erts_error.erl`. The test case `exception_SUITE:error_info/1` for Erts must also be updated to test error handling of the new BIFs.

**Comment by RaimoNiskanen** [comment]
@bjorng There is no extended error information for the already existing `erlang:set_cookie/2`...

**Comment by bjorng** [comment]
> @bjorng There is no extended error information for the already existing `erlang:set_cookie/2`...

Yes, I seemed to have missed that one. Could be because it does not entirely behave like a true BIF. It can raise either a `badarg` exception (for non-atoms) or `function_clause` (if distribution is not started). We should probably change that to always raise `badarg` on errors (with extended error information).

Do you want to fix `erlang:set_cookie/2` as part of this pull request while you are at it or do you want me to fix it?

**Comment by RaimoNiskanen** [comment]
> Do you want to fix erlang:set_cookie/2 as part of this pull request while you are at it or do you want me to fix it?

I am on it...

**Comment by RaimoNiskanen** [comment]
Now should extended error info be implemented

**Comment by RaimoNiskanen** [comment]
...also badarg from set_cookie and get_cookie instead of function clause

**Comment by RaimoNiskanen** [comment]
Added a disclaimer about "Security" not meaning the crypographical sense

**Comment by RaimoNiskanen** [comment]
Rebased to 'maint' (I first accidentally based this on 'master')

**Review by rickard-green — APPROVED** [review]

File: erts/doc/src/erl_cmd.xml [hunk 1]
```diff
@@ -629,6 +629,12 @@ $ <input>erl \
           <seemfa marker="erlang#set_cookie/2">
           <c>erlang:set_cookie/2</c></seemfa>.</p>
       </item>
+      <tag><c><![CDATA[-setcookie Node Cookie]]></c></tag>
+      <item>
+        <p>Sets the magic cookie for Node to <c><![CDATA[Cookie]]></c>; see
```
Comment (sverker) [inline erts/doc/src/erl_cmd.xml:634]: Should "Node" also be tagged with <c> like "Cookie".

File: erts/doc/src/erlang.xml [hunk 2]
```diff
@@ -2500,6 +2500,16 @@ uncompiled code with the same arity are mapped to the same list by
...
     </func>
 
+    <func>
+      <name name="get_cookie" arity="1" since="OTP @OTP-17538@"/>
+      <fsummary>Get the magic cookie for a node.</fsummary>
+      <desc>
+        <p>Returns the magic cookie for node <c><anno>Node</anno></c>
+          if the local node is alive,
+          otherwise the atom <c>nocookie</c>.</p>
+      </desc>
```
Comment (sverker) [inline erts/doc/src/erlang.xml:2510]: Is get_cookie(LocalNode) same as get_cookie()?

**Review by sverker — APPROVED** [review]

File: erts/doc/src/erlang.xml (hunk 2 above)
Comment (RaimoNiskanen) [reply erts/doc/src/erlang.xml:2510]: Yes.  `get_cookie()` is actually implemented with `get_cookie(node())`.

**Pull request merged**